pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -1391,7 +1402,9 @@ private void waitForCleanups(String broker, boolean excludeSystemTopics, int max
                     continue;
                 }
 
-                if (data.state() == Owned && broker.equals(data.dstBroker())) {
+                if (data.state() == Free) {
+                    futures.put(serviceUnit, handleFreeEvent(serviceUnit, data));
```

Comment:
why do we need to additionally handleFreeEvent here?

Reply:
It might not be needed here now. Let me double confirm it

Reply:
Reverted